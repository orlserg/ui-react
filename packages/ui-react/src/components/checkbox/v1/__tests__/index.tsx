import React from 'react';
import { create } from 'react-test-renderer';

import { Checkbox } from '..';

describe('Default', () => {
    test('Not checked', () => {
        const tree = create(
            <Checkbox
                name="default"
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('With ID', () => {
        const tree = create(
            <Checkbox
                id="custom-id"
                name="default"
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('Not checked & disabled', () => {
        const tree = create(
            <Checkbox
                name="default"
                disabled={true}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('Checked', () => {
        const tree = create(
            <Checkbox
                name="default"
                checked={true}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('Checked & disabled', () => {
        const tree = create(
            <Checkbox
                name="default"
                checked={true}
                disabled={true}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('Indeterminate', () => {
        const tree = create(
            <Checkbox
                name="default"
                indeterminate={true}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('Indeterminate & disabled', () => {
        const tree = create(
            <Checkbox
                name="default"
                indeterminate={true}
                disabled={true}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });
});

describe('Label', () => {
    const PROPS = {
        name : 'default',
        label: 'Checkbox label'
    };

    test('Not checked', () => {
        const tree = create(
            <Checkbox
                {...PROPS}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('Not checked & disabled', () => {
        const tree = create(
            <Checkbox
                {...PROPS}
                disabled={true}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('Checked', () => {
        const tree = create(
            <Checkbox
                {...PROPS}
                checked={true}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('Checked & disabled', () => {
        const tree = create(
            <Checkbox
                {...PROPS}
                checked={true}
                disabled={true}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('Indeterminate', () => {
        const tree = create(
            <Checkbox
                {...PROPS}
                indeterminate={true}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('Indeterminate & disabled', () => {
        const tree = create(
            <Checkbox
                {...PROPS}
                indeterminate={true}
                disabled={true}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });
});

describe('Label & Description', () => {
    const PROPS = {
        name       : 'default',
        label      : 'Checkbox label',
        description: 'Checkbox description'
    };

    test('Not checked', () => {
        const tree = create(
            <Checkbox
                {...PROPS}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('Not checked & disabled', () => {
        const tree = create(
            <Checkbox
                {...PROPS}
                disabled={true}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('Checked', () => {
        const tree = create(
            <Checkbox
                {...PROPS}
                checked={true}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('Checked & disabled', () => {
        const tree = create(
            <Checkbox
                {...PROPS}
                checked={true}
                disabled={true}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('Indeterminate', () => {
        const tree = create(
            <Checkbox
                {...PROPS}
                indeterminate={true}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('Indeterminate & disabled', () => {
        const tree = create(
            <Checkbox
                {...PROPS}
                indeterminate={true}
                disabled={true}
            />
        );

        expect(tree.toJSON()).toMatchSnapshot();
    });
});
