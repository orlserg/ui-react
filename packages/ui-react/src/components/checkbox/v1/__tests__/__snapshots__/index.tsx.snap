// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Checked & disabled 1`] = `
<label
  className="checkbox checkbox_disabled"
>
  <input
    aria-checked={true}
    checked={true}
    className="checkbox__input"
    disabled={true}
    name="default"
    role="checkbox"
    tabIndex={-1}
    type="checkbox"
  />
</label>
`;

exports[`Default Checked 1`] = `
<label
  className="checkbox"
>
  <input
    aria-checked={true}
    checked={true}
    className="checkbox__input"
    name="default"
    role="checkbox"
    tabIndex={0}
    type="checkbox"
  />
</label>
`;

exports[`Default Indeterminate & disabled 1`] = `
<label
  className="checkbox checkbox_disabled"
>
  <input
    aria-checked="mixed"
    className="checkbox__input checkbox__input_indeterminate"
    disabled={true}
    name="default"
    role="checkbox"
    tabIndex={-1}
    type="checkbox"
  />
</label>
`;

exports[`Default Indeterminate 1`] = `
<label
  className="checkbox"
>
  <input
    aria-checked="mixed"
    className="checkbox__input checkbox__input_indeterminate"
    name="default"
    role="checkbox"
    tabIndex={0}
    type="checkbox"
  />
</label>
`;

exports[`Default Not checked & disabled 1`] = `
<label
  className="checkbox checkbox_disabled"
>
  <input
    aria-checked={false}
    className="checkbox__input"
    disabled={true}
    name="default"
    role="checkbox"
    tabIndex={-1}
    type="checkbox"
  />
</label>
`;

exports[`Default Not checked 1`] = `
<label
  className="checkbox"
>
  <input
    aria-checked={false}
    className="checkbox__input"
    name="default"
    role="checkbox"
    tabIndex={0}
    type="checkbox"
  />
</label>
`;

exports[`Default With ID 1`] = `
<label
  className="checkbox"
  id="custom-id"
>
  <input
    aria-checked={false}
    className="checkbox__input"
    id="custom-id-input"
    name="default"
    role="checkbox"
    tabIndex={0}
    type="checkbox"
  />
</label>
`;

exports[`Label & Description Checked & disabled 1`] = `
<label
  className="checkbox checkbox_disabled checkbox_description"
>
  <input
    aria-checked={true}
    checked={true}
    className="checkbox__input"
    disabled={true}
    name="default"
    role="checkbox"
    tabIndex={-1}
    type="checkbox"
  />
  Checkbox label
  <span
    className="checkbox__description"
  >
    Checkbox description
  </span>
</label>
`;

exports[`Label & Description Checked 1`] = `
<label
  className="checkbox checkbox_description"
>
  <input
    aria-checked={true}
    checked={true}
    className="checkbox__input"
    name="default"
    role="checkbox"
    tabIndex={0}
    type="checkbox"
  />
  Checkbox label
  <span
    className="checkbox__description"
  >
    Checkbox description
  </span>
</label>
`;

exports[`Label & Description Indeterminate & disabled 1`] = `
<label
  className="checkbox checkbox_disabled checkbox_description"
>
  <input
    aria-checked="mixed"
    className="checkbox__input checkbox__input_indeterminate"
    disabled={true}
    name="default"
    role="checkbox"
    tabIndex={-1}
    type="checkbox"
  />
  Checkbox label
  <span
    className="checkbox__description"
  >
    Checkbox description
  </span>
</label>
`;

exports[`Label & Description Indeterminate 1`] = `
<label
  className="checkbox checkbox_description"
>
  <input
    aria-checked="mixed"
    className="checkbox__input checkbox__input_indeterminate"
    name="default"
    role="checkbox"
    tabIndex={0}
    type="checkbox"
  />
  Checkbox label
  <span
    className="checkbox__description"
  >
    Checkbox description
  </span>
</label>
`;

exports[`Label & Description Not checked & disabled 1`] = `
<label
  className="checkbox checkbox_disabled checkbox_description"
>
  <input
    aria-checked={false}
    className="checkbox__input"
    disabled={true}
    name="default"
    role="checkbox"
    tabIndex={-1}
    type="checkbox"
  />
  Checkbox label
  <span
    className="checkbox__description"
  >
    Checkbox description
  </span>
</label>
`;

exports[`Label & Description Not checked 1`] = `
<label
  className="checkbox checkbox_description"
>
  <input
    aria-checked={false}
    className="checkbox__input"
    name="default"
    role="checkbox"
    tabIndex={0}
    type="checkbox"
  />
  Checkbox label
  <span
    className="checkbox__description"
  >
    Checkbox description
  </span>
</label>
`;

exports[`Label Checked & disabled 1`] = `
<label
  className="checkbox checkbox_disabled"
>
  <input
    aria-checked={true}
    checked={true}
    className="checkbox__input"
    disabled={true}
    name="default"
    role="checkbox"
    tabIndex={-1}
    type="checkbox"
  />
  Checkbox label
</label>
`;

exports[`Label Checked 1`] = `
<label
  className="checkbox"
>
  <input
    aria-checked={true}
    checked={true}
    className="checkbox__input"
    name="default"
    role="checkbox"
    tabIndex={0}
    type="checkbox"
  />
  Checkbox label
</label>
`;

exports[`Label Indeterminate & disabled 1`] = `
<label
  className="checkbox checkbox_disabled"
>
  <input
    aria-checked="mixed"
    className="checkbox__input checkbox__input_indeterminate"
    disabled={true}
    name="default"
    role="checkbox"
    tabIndex={-1}
    type="checkbox"
  />
  Checkbox label
</label>
`;

exports[`Label Indeterminate 1`] = `
<label
  className="checkbox"
>
  <input
    aria-checked="mixed"
    className="checkbox__input checkbox__input_indeterminate"
    name="default"
    role="checkbox"
    tabIndex={0}
    type="checkbox"
  />
  Checkbox label
</label>
`;

exports[`Label Not checked & disabled 1`] = `
<label
  className="checkbox checkbox_disabled"
>
  <input
    aria-checked={false}
    className="checkbox__input"
    disabled={true}
    name="default"
    role="checkbox"
    tabIndex={-1}
    type="checkbox"
  />
  Checkbox label
</label>
`;

exports[`Label Not checked 1`] = `
<label
  className="checkbox"
>
  <input
    aria-checked={false}
    className="checkbox__input"
    name="default"
    role="checkbox"
    tabIndex={0}
    type="checkbox"
  />
  Checkbox label
</label>
`;
